presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -905,6 +934,77 @@ public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle
         metastore.dropTable(session, handle.getSchemaName(), handle.getTableName());
     }
 
+    @Override
+    public ConnectorAnalyzeTableHandle beginAnalysis(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        verifyJvmTimeZone();
+        HiveTableHandle handle = (HiveTableHandle) tableHandle;
+        SchemaTableName tableName = handle.getSchemaTableName();
+
+        metastore.getTable(tableName.getSchemaName(), tableName.getTableName())
```

Comment:
you could now simply make this condition explicit:
```
if (!metastore.getTable(...).isPresent()) {
  throw new TableNotFoundException(tableName);
}
```

Reply:
Should we add a test now that there is not writable check?

Also `checkTableIsWritable` apart from external table check also called: `HiveWriteUtils#checkWritable` which consist of:
```
// verify online
// verify not read only
// verify skew info
```
Can we skip all of those too?

Reply:
@jessesleeping Could you verify Presto ANALYZE behavior vs Hive (automated product test is preferable), particularly
* Can one analyze table in Hive that is offline?
* Can one analyze table in Hive that is read only?
* Can one analyze table in Hive that is skewed?

Reply:
We probably should be consistent with Hive within Hive realm. I would expect that Hive also allows to run Analyze for offline/read only/skewed tables but we should make sure that is the case. 
If Hive disallows ANALYZE for "read only" tables for instance, then data administrator might expect the same behavior from Presto.

Reply:
Exactly. People used to hive behavior and permissions will expect Presto to follow suite.
In a hipothetical situation, if Hive allowed ANALYZE-ing tables that user does not have read permission to, we should do the same (in Hive connector).

> offline/readonly/skewed tables? (btw, what are the first two?)

I don't know, but there is `org.apache.hadoop.hive.metastore.ProtectMode` ..